### PR TITLE
`web.run_async` as alternative to `web.Batch`

### PR DESCRIPTION
**Pull request by tylerflex**
instead of 

```py
batch = web.Batch(simulations=sims)
batch_results = batch.run(path_dir="data")
```

you can use `asyncio` [package](https://realpython.com/async-io-python/) to do
```py
batch_results = await web.async_run(sims)
```

and the simulations will run asynchronously 

uses an `asyncio.Queue` under the hood.

@joamatab

### CODE REVIEW

**Comment by tylerflex** [comment]
Docs PR: https://github.com/flexcompute-readthedocs/tidy3d-docs/pull/126

**Review by shashwat-sh — APPROVED** [review]
Looks good! Some observations from trying it out:
- If I forget to add `await`, it looks like nothing happens - there's no error or warning, which might be confusing for a user? It's pretty clearly documented that `await` must be used, so maybe it's okay to let the user figure out why nothing is happening.
- In my notebook, the output cell got saturated pretty fast (image attached). Is there a way to just show a "started" and "finished" status?

I don't think these are critical, so feel free to ignore if you think the current behaviour is okay.

image

File: tidy3d/web/queue.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        Base directory where data will be downloaded, by default current working directory.
+    callback_url : str = None
+        Http PUT url to receive simulation finish event. The body content is a json file with
+        fields ``{'id', 'status', 'name', 'workUnit', 'solverVersion'}``.
+    num_workers: int = None
+        Number of tasks to submit at once in a batch, if None, will run all at the same time.
+
+    Note
+    ----
+    It is necessary to "await" the return of this function. ie ``await run_async()``.
```
[the remark this answers is not in the document]
Comment (tylerflex) [reply tidy3d/web/queue.py:37]: Hm, so I just tried a couple things:
* making `run_async` a normal function (not `async_def`) and got errors.
* awaiting the return value in `run_async`, which seemed to still require `await` when used in the notebook.
So I don't understand completely but I think we need to live with `await` when using this feature.

**Comment by tylerflex** [comment]
> If I forget to add await, it looks like nothing happens 

I dont know if there's a way to let them know that `await` is needed directly in the code, unfortunately.. I've looked into it and can't find anything. What gets returned in this case is just the "`coroutine`" which can then be "awaited" later but I figure anyone using this feature should read the tutorial I linked to get a better idea so they are familiar with it.

>  Is there a way to just show a "started" and "finished" status?

Not until we are able to turn off the webapi verbosity, which is still under review #694. After this, I think I will make another PR to make the underlying functions `verbose=False` and add a simpler progress bar just for the workers.

File: tidy3d/web/queue.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shashwat-sh) [reply tidy3d/web/queue.py:37]: Ok, yeah I don't think it's a big deal. If there's a way to detect if `await` was not used and then warn the user, that could be useful, but it sounds like that might not be trivial / worth it.

**Comment by shashwat-sh** [comment]
> > If I forget to add await, it looks like nothing happens
> 
> I dont know if there's a way to let them know that `await` is needed directly in the code, unfortunately.. I've looked into it and can't find anything. What gets returned in this case is just the "`coroutine`" which can then be "awaited" later but I figure anyone using this feature should read the tutorial I linked to get a better idea so they are familiar with it.
> 
> > Is there a way to just show a "started" and "finished" status?
> 
> Not until we are able to turn off the webapi verbosity, which is still under review #694. After this, I think I will make another PR to make the underlying functions `verbose=False` and add a simpler progress bar just for the workers.

Ah ok, sounds good.

**Pull request merged**